src/components/category/CategoryList.js

```diff
@@ -0,0 +1,81 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { CategoryContext } from './CategoryProvider';
+import { BsFillGearFill, BsFillTrashFill } from 'react-icons/bs';
+import "./Category.css";
+
+export const CategoryList = () => {
+    const { getAllCategories, addCategory, categories, updateCategory, deleteCategory } = useContext(CategoryContext)
+    const [newCategoryName, setNewCategoryName] = useState({})
+    const history = useHistory()
+
+    useEffect(() => {
+        getAllCategories()
+    }, [])
+
+    const handleInputChange = (event) => {
+        const newCat = {}
+        newCat.label = event.target.value
+        setNewCategoryName(newCat)
+    }
+
+    const handleSaveCategory = () => {
+        addCategory({
+            "label": newCategoryName.label
+        })
+        .then(() => history.push("/categories"))
+    }
+
+    const removeCategory = (catId) => {
+        // TODO: needs to be within a modal box
+        deleteCategory(parseInt(catId))
+    }
+
+    const editCategory = (catObj) => {
+        // TODO: needs to be invoked on click within modal box with input
+        updateCategory({
+            "id": parseInt(catObj.id),
+            "label": newCategoryName.label
+        })
+        .then(() => history.push("/categories"))
+    }
+
+    return (
+        <>
+            <div className="categories--flex--outer">
+                <div className="categories__list">
+                    <h3>Categories</h3>
+                    {
+                        categories.map(catObj => {
```

Comment:
This could be an implicit `return` and the only reason it matters here is this:

- this `.map` is happening **inside the render** (inside the returned JSX) so it shouldn't be "doing" anything but rendering
- by **never** using the explicit `return` from an inline function in your render, you can visually confirm that there is no funny business being done inside the render
- just to reiterate -- the code is fine and very readable as-is. I just think "implicit returns only within a render" is a good way of applying a very useful constraint (no non-render code inside the render).

Reply:
Fixed in next PR